zebra/rt_netlink.c

```diff
@@ -246,6 +246,42 @@ static vrf_id_t vrf_lookup_by_table(uint32_t table_id, ns_id_t ns_id)
 	return VRF_DEFAULT;
 }
 
+/**
+ * @parse_encap_mpls() - Parses encapsulated mpls attributes
+ * @tb:         Pointer to rtattr to look for nested items in.
+ * @labels:     Pointer to store labels in.
+ *
+ * Return:      Number of mpls labels found.
+ */
+static int parse_encap_mpls(struct rtattr *tb, mpls_label_t *labels)
+{
+	struct rtattr *tb_encap[MPLS_IPTUNNEL_MAX + 1];
+	int num_labels = 0;
+	uint32_t ttl = 0;
+	uint32_t bos = 0;
+	uint32_t exp = 0;
+	mpls_label_t label = 0;
+
+	netlink_parse_rtattr_nested(tb_encap, MPLS_IPTUNNEL_MAX, tb);
+	while (!bos) {
+		mpls_lse_t lse =
+			*(mpls_label_t *)RTA_DATA(tb_encap[MPLS_IPTUNNEL_DST]);
+		mpls_lse_decode(lse, &label, &ttl, &exp, &bos);
```

Comment:
This is an infinite loop when the route has more than one label. `lse` always points to the first label (top of the stack).

Reply:
I believe this has been addressed